std/algorithm/iteration.d

```diff
@@ -836,7 +838,7 @@ template each(alias pred = "a")
 
     enum isForeachBinaryIterable(R) =
         is(typeof((R r) {
-            foreach (i, ref a; r)
+            foreach (ref i, ref a; r)
```

Comment:
A quick hack might be to check whether it's a tuple
 -> if `no` then no ref
-> if `yes` check for `r.front.index` ... (enumerate tuples are named)